tests/extension/server/runmeServer.test.ts

```diff
@@ -38,6 +45,28 @@ suite('Runme server spawn process', () => {
         await expect(server.launch()).rejects.toBeInstanceOf(RunmeServerError)
         expect(serverLaunchSpy).toBeCalledTimes(3)
     })
+
+    test('Should increment until port is available', async () => {
+      const server = new Server(
+        '/Users/user/.vscode/extension/stateful.runme',
+        {
+          retryOnFailure: true,
+          maxNumberOfIntents: 2,
+        }
+      )
+
+      vi.mocked(fs.access).mockResolvedValueOnce()
+      // @ts-expect-error
+      vi.mocked(fs.stat).mockResolvedValueOnce({
+        isFile: vi.fn().mockReturnValue(true)
+      })
```

Comment:
Not sure what the TS error is but I usually just cast things to any
```suggestion
      vi.mocked(fs.stat).mockResolvedValueOnce({
        isFile: vi.fn().mockReturnValue(true)
      } as any)
```

Reply:
TS error is because `fs.stat` expects a much larger struct. But yeah, casting as any here is better!